🐛 BUG: `onlineornot checks delete` errors are incomprehensible

### What version of `onlineornot` are you using?

1.0.1

### What operating system are you using?

Linux

### Describe the Bug

```
 ✅ onlineornot 1.0.1 
---------------------

✘ [ERROR] A request to the OnlineOrNot API (/checks/26pJJpKZ) failed.


If you think this is a bug then please create an issue at https://github.com/OnlineOrNot/onlineornot/issues/new/choose
```

--- a/.changeset/friendly-dogs-smile.md
+++ b/.changeset/friendly-dogs-smile.md
@@ -1,0 +1,5 @@
+---
+"onlineornot": patch
+---
+
+Fix `checks delete` error handling to show clear "not found" message instead of generic error

--- a/packages/onlineornot/src/checks/delete.ts
+++ b/packages/onlineornot/src/checks/delete.ts
@@ -33,11 +33,13 @@ export async function handler(args: StrictYargsOptionsToInterface<typeof options
 		if (errorWithCode.code === 10009 || errorWithCode.code === 10001) {
 			// 10009 = NOT_FOUND, 10001 = ISSUE_FETCHING_DATA (legacy, also means not found)
 			return logger.error(`Check "${args.id}" not found.`);
-		} else if (errorWithCode.code === 10002) {
+		} else if (errorWithCode.code === 10002 || errorWithCode.code === 10011) {
+			// 10002 = UNAUTHENTICATED, 10011 = UNAUTHENTICATED_TOKEN
 			return logger.error(
 				"You are not authenticated.\nPlease check your token with `onlineornot whoami` and try again.",
 			);
-		} else if (errorWithCode.code === 10003) {
+		} else if (errorWithCode.code === 10003 || errorWithCode.code === 10012) {
+			// 10003 = UNAUTHORIZED, 10012 = INSUFFICIENT_PERMISSIONS
 			return logger.error(
 				"Your API token isn't allowed to delete checks.\nPlease check your token with `onlineornot whoami` and try again.",
 			);

--- a/packages/onlineornot/src/checks/delete.ts
+++ b/packages/onlineornot/src/checks/delete.ts
@@ -23,9 +23,29 @@ export async function handler(args: StrictYargsOptionsToInterface<typeof options
 		await printBanner();
 	}
 	await verifyToken();
-	await fetchResult(`/checks/${args.id}`, {
-		method: "DELETE",
-	});
+
+	try {
+		await fetchResult(`/checks/${args.id}`, {
+			method: "DELETE",
+		});
+	} catch (err) {
+		const errorWithCode = err as { code?: number; notes?: { text: string }[] };
+		if (errorWithCode.code === 10009 || errorWithCode.code === 10001) {
+			// 10009 = NOT_FOUND, 10001 = ISSUE_FETCHING_DATA (legacy, also means not found)
+			return logger.error(`Check "${args.id}" not found.`);
+		} else if (errorWithCode.code === 10002) {
+			return logger.error(
+				"You are not authenticated.\nPlease check your token with `onlineornot whoami` and try again.",
+			);
+		} else if (errorWithCode.code === 10003) {
+			return logger.error(
+				"Your API token isn't allowed to delete checks.\nPlease check your token with `onlineornot whoami` and try again.",
+			);
+		} else {
+			return logger.error(err);
+		}
+	}
+
 	if (args.json) {
 		logger.log(JSON.stringify({ id: args.id }, null, "  "));
 	} else {

